File: backend/src/main/java/com/smartcampus/backend/dto/weather/WeatherSummaryDTO.java
package com.smartcampus.backend.dto.weather;

import lombok.*;

import java.util.List;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WeatherSummaryDTO {


    private Double latitude;

    private Double longitude;


    private CurrentWeatherDTO current;


    private List<HourlyPointDTO> hourlyToday;


    private List<DailyWeatherDTO> pastDaily;


    private List<DailyWeatherDTO> forecastDaily;

}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/weather/WeatherSummaryDto.kt
package com.ObservatoireCampus.mobile.model.weather

data class WeatherSummaryDto(
    val latitude: Double?,
    val longitude: Double?,
    val current: CurrentWeatherDto?,
    val hourlyToday: List<HourlyWeatherPointDto> = emptyList(),
    val pastDaily: List<DailyWeatherDto> = emptyList(),
    val forecastDaily: List<DailyWeatherDto> = emptyList()
)